### DISCUSSION TITLE
Reduce/Remove network calls from unittests

### DISCUSSION THREAD

**Discussion by ericpien** [ranaroussi/yfinance#2136] [Development]
Moving discussion from https://github.com/ranaroussi/yfinance/issues/2093#issuecomment-2427341424 to not spam everyone involved.

@ValueRaider If we mock the network calls in the unittests, we could reduce the time required for testing and avoid spamming yahoo (i.e. save the response as `.pkl` and read the mock data instead of fetching).

I think an advantage of keeping them a network call is that we could catch changes in yahoo finance API responses. However, that isn't really the purpose of unittests and we wouldn't catch such changes in the absence of running unittests when pushing to `main/dev` anyway.

Do you think this is worth trying?

**Comment by ValueRaider** [comment]
> I think an advantage of keeping them a network call is that we could catch changes in yahoo finance API responses.

Not really. Changes can be geographical, or only when Yahoo under high load so temporary.

> Do you think this is worth trying?

Sounds good and can improve some tests. Can you update/prototype one test to show what it looks like?
